types/index.d.ts

```diff
@@ -131,6 +160,11 @@ declare module '@azure/functions' {
          * Route parameter keys and values.
          */
         params: HttpRequestParams;
+        /**
+         *  Object representing logged-in user, either through
+         *  AppService/Functions authentication, or SWA Authentication
+         */
+        user?: HttpRequestUser | null;
```

Comment:
We have to be careful about `undefined` vs `null`, especially because we do not have linting properly set up to cover this area yet. The differences are subtle, but definitely there. In this case, I would get rid of the question mark.
```suggestion
        user: HttpRequestUser | null;
```

`undefined` means we have not figured out what the value of user is yet, whereas `null` means we have figured it out and there is no user. We should never return `undefined` here and the types should reflect that